src/v4.rs

```diff
@@ -23,12 +23,8 @@ impl Uuid {
     ///
     /// [`rand`]: https://crates.io/crates/rand
     pub fn new_v4() -> Self {
-        use rand::RngCore;
-
-        let mut rng = rand::thread_rng();
         let mut bytes = [0; 16];
-
-        rng.fill_bytes(&mut bytes);
+        getrandom(&mut bytes).expect("RNG failure!");
```

Comment:
Another issue: you are using `.expect(...)`. I assume it is to maintain the signature of the constructor. But since a `1.0` would be a major release anyways, we can break it. You may have to adjust the error itself to support this though.

Reply:
I had a go at it. Could you please see if the changes I made are along the lines of what you were thinking?